Migration.md

```diff
@@ -32,6 +32,11 @@ release will remove the deprecated code.
     + Added `Scene::SetCameraPassCountPerGpuFlush`. Setting this value to 0 forces legacy behavior which eases porting.
     + Systems that rely on Graphics components like particle FXs and postprocessing are explicitly affected by Scene's Pre/PostRender. Once `Scene::PostRender` is called, the particle FXs' simulation is moved forward, as well as time values sent to postprocessing shaders. In previous ign-rendering versions each `Camera::Render` call would move the particle simulation forward, which could cause subtle bugs or inconsistencies when Cameras were rendering the same frame from different angles. Setting SetCameraPassCountPerGpuFlush to 0 will also cause these subtle bugs to reappear.
 
+1. **Visual.hh** and **Node.hh**
+    + `*UserData` methods and the `Variant` type alias have been moved from the `Visual` class to the `Node` class.
+    `Node::UserData` now returns no data for keys that don't exist (prior to Rendering 6.x, if
+    `Visual::UserData` was called with a key that doesn't exist, an `int` was returned by default).
```

Comment:
an `int` with a value of `0`?

Reply:
Is it safe to always assume "value of `0`"? The [`std::variant` documentation](https://en.cppreference.com/w/cpp/utility/variant) states, _"a default-constructed variant holds a value of its first alternative"_, which makes me think that the `int` (the first alternative in this case) is also default-initialized. If that's the case, there's technically no guarantee that the `int` will be initialized to `0`. But maybe the variant initializes alternatives to a specific value? I'm not sure.

Reply:
I'm not sure too. I saw the comment https://github.com/ignitionrobotics/ign-rendering/pull/334#issuecomment-882382753 that says it'll always return 0. Maybe that's not always true. We could leave the doc here as is if we don't know the answer.

Reply:
Okay, I'll leave it as-is because I think it's `0` most of the time, but I'm not sure if that's a guarantee.